library/network_connections.py

```diff
@@ -976,6 +976,16 @@ def connection_create(self, connections, idx, connection_current=None):
                     else:
                         s_ethtool.option_set_uint32(nm_ring, setting)
 
+            # * When user never asked about ethtool configuration, network-role
+            #   will generate an all-default `NM.SettingEthtool` and pass it to
+            #   NetworkManager daemon. But NetworkManager discard it when saving to
+            #   ifcfg plugin as ifcfg plugin will not keep empty ethtool option.
+            # * The follow up `NM.SimpleConnection.compare()` will indicate
+            #   configuration changed as new ethtool setting found.
+            # * To workaround this, we just remove the all-default NM.SettingEthtool.
```

Comment:
Here are some ideas to improve the wording. I am not 100% happy with my attempt, though.

```suggestion
            # * When users did not specify any ethtool configuration, this module
            #   will generate an default `NM.SettingEthtool` object and pass it to
            #   NetworkManager. But NetworkManager cannot serialize this when using the ifcfg
            #   plugin but treats this as no `NM.SettingEthtool` object.
            # * The following `NM.SimpleConnection.compare()` will therefore identify a
            #   difference in the configuration.
            # * To workaround this, remove the default NM.SettingEthtool object.
```